enhancements/network/multi-network-vrfs.md

```diff
@@ -0,0 +1,924 @@
+
+#### OpenShift Router/Ingress
+
+One of the goals of this effort is to allow ingress services to pods that live on primary user-defined networks.
+
+OpenShift router is deployed in two different ways:
+
+* On-prem: deployed as a host networked pod running haproxy, using a keepalived VIP.
+* Cloud: deployed as an OVN networked pod running haproxy, fronted by a cloud external load balancer which forwards
+traffic to a nodeport service with ETP Local.
+
+OpenShift router is implemented as HAProxy, which is capable of terminating TLS or allowing TLS passthrough and
+forwarding HTTP connections directly to endpoints of services.
+
+##### On-Prem
+
+Due to previously listed constraints in the Host Networked Pods section, host networked pods are unable to talk directly
+to OVN networked pods unless bound to the VRF itself. HAProxy needs to communicate with potentially many VRFs, so
+therefore this functionality does not work. However, a host networked pod is capable of reaching any service CIDR. In
+order to solve this problem, openshift-ingress-operator will be modified so that when it configures HAProxy, it will
+check if the service belongs to a user-defined network, and then use the service CIDR as the forwarding path.
+
+##### Cloud
+
+In the cloud case, the ovn-networked HAProxy pod cannot reach pods on other networks. Like the On-Prem case, HAProxy
+will be modified for user-defined networks to forward to the service CIDR rather than the endpoint directly. The caveat
+here is that an OVN networked pod only has service access to:
+
+1. CoreDNS and Kube API via the cluster default network
+2. Any service that resides on the same namespace as the pod
+
+To solve this problem, the OpenShift ingress namespace will be given permission to access any service, by forwarding
+service access via mp0, and allowing the access with ipTables rules. In OVN-Kubernetes upstream, we can allow a
+configuration to list certain namespaces that are able to have access to all services. This should be reserved for
+administrators to configure.
+
+A diagram of how the Cloud traffic path will work:
+
+![OpenShift Router Multi Network](images/openshift-router-multi-network.svg)
+
+##### Limitations
+
+With OpenShift Router, stickiness is used to ensure that traffic for a session always reaches the same endpoint. For TLS
+passthrough, HAProxy uses the source IP address as that's all it has (it's not decrypting the connection, so it can't
+look at cookies, for example).  Otherwise, it uses a cookie.
+
+By changing the behavior of HAProxy to forward to a service CIDR instead of an endpoint directly, we cannot accommodate
+using cookies for stickiness. However, a user can configure the service with sessionAffinity, which will cause OVN to
+use stickiness by IP. Therefore, users who wish to use OpenShift router with user-defined networks will be limited to
+only enabling session stickiness via client IP.
+
```

Comment:
```suggestion
The live migration feature at layer2 described here will work only at interconnect since there is no mac learning between zones, so we can have the same extra address at the route type LSP on switches.
```

Reply:
@qinqon : IIUC this whole feature is only limited to interconnect mode... right @trozet ? are we planning to also support this on default mode for every feature?

Reply:
Hm, then I think that should be clearly stated in the limitations section. 

I had assumed we wanted as many things as possible to be available on non-IC deployments.

Reply:
the limitations section here is a sub-heading for ingress. I'll add your text somewhere that makes more sense.